LibraryServices/Services/Library.cs

```diff
@@ -0,0 +1,300 @@
+        /// Return All Books
+        /// </summary>
+        public IEnumerable<Book> GetAllBook()
+        {
+            foreach (Book book in _booksInLibrary)
+            {
+                yield return book;
+            }
+        }
+
+        /// <summary>
+        /// Return Book by Id
+        /// </summary>
+        /// <param name="id">Book's Id</param>
+        /// <returns></returns>
+        public Book GetBook(int id)
+        {
+            return _booksInLibrary.Find(Book => Book.Id == id);
+        }
+
+        /// <summary>
+        /// Update Book in collection
+        /// </summary>
+        /// <param name="id">Book's Id</param>
+        /// <param name="book">New Book info</param>
+        /// <returns>Updated Book</returns>
+        public Book UpdateBook(int id, Book book)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == id);
+            if (book != null)
+            {
+                bookToUpdate.Name = book.Name;
+                bookToUpdate.AuthorId = book.AuthorId;
+                bookToUpdate.Year = book.Year;
+            }
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Delete Author from Book
+        /// </summary>
+        /// <param name="bookId">Book's Id</param>
+        /// <returns>Updated Book without author</returns>
+        public Book DeleteAutorFromBook(int id)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == id);
+            bookToUpdate.AuthorId = null;
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Add Author To Book
+        /// </summary>
+        /// <param name="bookId">Book's Id</param>
+        /// <param name="authorId">Author's Id</param>
+        /// <returns>New Book with added Author</returns>
+        public Book AddAuthorToBook(int bookId, int authorId)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == bookId);
+            bookToUpdate.AuthorId = authorId;
+            return bookToUpdate;
+        }
+
+        /// <summary>
+        /// Adds the genre to book.
+        /// </summary>
+        /// <param name="book_id">The book identifier.</param>
+        /// <param name="genre_id">The genre identifier.</param>
+        /// <returns>Is added new link.</returns>
+        public bool AddGenreToBook(int book_id, int genre_id)
+        {
+            bool result = false;
+
+            if (GetBook(book_id) != null && _genres.Any((genre) => genre.Id == genre_id))
+            {
+                _bookAndGenresConnection.Add(new KeyValuePair<int, int>(book_id, genre_id));
+                result = true;
+            }
+            return result;
+        }
+
+        #endregion
+
+        #region AuthorRegion
+
+        /// <summary>
+        /// Return All Authors
+        /// </summary>
+        public IEnumerable<Author> GetAllAuthors()
+        {
+            foreach (Author author in _authors)
```

Comment:
I'm starting to think that I don't know something. And you have some big reason to write 3 lines instead of a simple one with the return.